packages/navigator/src/browser/open-editors-widget/navigator-open-editors-widget.tsx

```diff
@@ -0,0 +1,178 @@
+import { EditorPreviewWidget } from '@theia/editor-preview/lib/browser';
+
+const CLOSE_ICON_WIDTH = 16;
+
+export const OPEN_EDITORS_PROPS: TreeProps = {
+    ...defaultTreeProps,
+    virtualized: false,
+    contextMenuPath: OPEN_EDITORS_CONTEXT_MENU,
+};
+@injectable()
+export class OpenEditorsWidget extends FileTreeWidget {
+    static ID = 'theia-open-editors-widget';
+    static LABEL = 'Open Editors';
+
+    static PREFIX_ICON_CLASS = 'open-editors-prefix-icon';
+
+    @inject(ApplicationShell) protected readonly applicationShell: ApplicationShell;
+
+    static createContainer(parent: interfaces.Container): Container {
+        const child = createFileTreeContainer(parent);
+
+        child.unbind(FileTreeModel);
+        child.bind(OpenEditorsModel).toSelf();
+        child.rebind(TreeModel).toService(OpenEditorsModel);
+
+        child.unbind(FileTreeWidget);
+        child.bind(OpenEditorsWidget).toSelf();
+
+        child.rebind(TreeProps).toConstantValue(OPEN_EDITORS_PROPS);
+
+        child.bind(OpenEditorsTreeDecoratorService).toSelf().inSingletonScope();
+        child.rebind(TreeDecoratorService).toService(OpenEditorsTreeDecoratorService);
+        return child;
+    }
+
+    static createWidget(parent: interfaces.Container): OpenEditorsWidget {
+        return OpenEditorsWidget.createContainer(parent).get(OpenEditorsWidget);
+    }
+
+    constructor(
+        @inject(TreeProps) readonly props: TreeProps,
+        @inject(OpenEditorsModel) readonly model: OpenEditorsModel,
+        @inject(ContextMenuRenderer) protected readonly contextMenuRenderer: ContextMenuRenderer
+    ) {
+        super(props, model, contextMenuRenderer);
+    }
+
+    @postConstruct()
+    init(): void {
+        super.init();
+        this.id = OpenEditorsWidget.ID;
+        this.title.label = OpenEditorsWidget.LABEL;
+        this.addClass(OpenEditorsWidget.ID);
+        this.update();
+    }
+    protected activeTreeNodePrefixElement: string | undefined | null;
+
+    protected renderNode(node: OpenEditorNode, props: NodeProps): React.ReactNode {
+        if (!TreeNode.isVisible(node)) {
+            return undefined;
+        }
+        const attributes = this.createNodeAttributes(node, props);
+        const content = <div className={TREE_NODE_CONTENT_CLASS}
+            onMouseEnter={this.handleMouseEnter}
+            onMouseLeave={this.handleMouseLeave}>
+            {this.renderPrefixIcon(node)}
+            {this.decorateIcon(node, this.renderIcon(node, props))}
+            {this.renderCaptionAffixes(node, props, 'captionPrefixes')}
+            {this.renderCaption(node, props)}
+            {this.renderCaptionAffixes(node, props, 'captionSuffixes')}
+            {this.renderTailDecorations(node, props)}
+        </div>;
+        return React.createElement('div', attributes, content);
+    }
+
+    protected handleMouseEnter = (e: React.MouseEvent<HTMLDivElement>) => this.doHandleMouseEnter(e);
```

Comment:
Could the logic handled by this enter logic be shifted onto CSS? The way font-awesome and codicon tend to work is by putting some random unicode in a `:after` and then assigning that unicode to the particular icon in their font definition, so it might end up being something like this:

```CSS
.tree-node .left-slot {
    /* normally nothing, so probably no rule */
}

.tree-node.dirty .left-slot:after {
    content: /* unicode corresponding to closed circle in codicon font */
}

.tree-node:hover .left-slot {
   content: /* unicode corresponding to exit icon in codicon font */
}
```

And since you've already added the codicon class, you should get the correct icon.

Reply:
Good catch, I knew there must be a better way to do this. I've gone with your second recommendation since I wasn't able to get the unicode approach working